src/components/expression/expression-loader.js

```diff
@@ -133,40 +148,103 @@ genotet.ExpressionLoader.prototype.loadExpressionMatrix_ = function(fileName,
       });
     this.data.lowerConditionNames = lowerConditionNames;
 
-    this.loadTfaProfile_(this.data.tfa.fileName, this.data.matrix.geneNames,
-      this.data.matrix.conditionNames);
+    if (this.data.profile.geneNames.length != 0) {
+      this.loadProfile(fileName, this.data.profile.geneNames,
+        data.conditionNames, false);
+    }
+    if (this.data.tfa.geneNames.length != 0) {
+      this.loadTfaProfile(fileName, this.data.tfa.geneNames,
+        data.conditionNames, false);
+    }
   }.bind(this), 'cannot load expression matrix');
 };
 
 /**
- * Loads the TFA for selected genes.
+ * Loads the profile for selected genes.
  * @param {string} fileName Name of the expression matrix.
- * @param {!Array<string>} geneNames Names for gene selection.
+ * @param {!Array<string>} geneNames Name for gene selection.
  * @param {!Array<string>} conditionNames Names for experiment condition
  *      selection.
- * @private
+ * @param {boolean} isAddProfile Whether it is adding a new profile.
  */
-genotet.ExpressionLoader.prototype.loadTfaProfile_ = function(fileName,
-    geneNames, conditionNames) {
-  var tfaParams = {
-    type: genotet.expression.QueryType.TFA_PROFILE,
-    fileName: genotet.data.tfaFileName,
-    geneNames: geneNames,
-    conditionNames: conditionNames
+genotet.ExpressionLoader.prototype.loadProfile =
+  function(fileName, geneNames, conditionNames, isAddProfile) {
+    var params = {
+      type: genotet.expression.QueryType.PROFILE,
+      fileName: fileName,
+      geneNames: geneNames,
+      conditionNames: conditionNames
+    };
+    this.get(genotet.data.serverURL, params, function(data) {
+      // Store the last applied data selectors.
+      if (data.geneNames.length == 0 || data.conditionNames.length == 0) {
```

Comment:
?? Isn't that just what you want? I suppose you want to check if length is zero??